### PR TITLE
NPC basic state

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: -

* **Server Mode**: -

* **Description of Pull Request**: 

Suggestion to allow user to set the npc basic state in the header instead of the (too many times defined) 'OnInit' label.

### CODE REVIEW

File: src/map/npc.cpp [hunk 1]
```diff
@@ -5507,16 +5528,21 @@ int npc_parsesrcfile(const char* filepath)
...
 			count2 = sscanf(w1,"%15[^,],%6hd,%6hd[^,]",mapname,&x,&y);
-			
+
 			if ( count2 < 1 ) {
 				ShowError("npc_parsesrcfile: Invalid script definition in file '%s', line '%d'. Skipping line...\n * w1=%s\n * w2=%s\n * w3=%s\n * w4=%s\n", filepath, strline(buffer,p-buffer), w1, w2, w3, w4);
-				if (strcasecmp(w2,"script") == 0 && count > 3) {
+				if (has_script) {
 					if ((p = npc_skip_script(p,buffer,filepath)) == NULL)
 						break;
 				}
```
Comment (Daegaladh) [inline src/map/npc.cpp:5548]: While you're at it, could you clean up this checks a bit?, get rid of all those curly braces and change the nulls to nullptr, so it'd be a bit easier to read :P

**Comment by mmo4th** [comment]
Hello, any sample scripts?

**Comment by Atemo** [comment]
```
lighthalzen,306,267,5	script(CLOAKED)	Skia#ep162_04	4_EP16_SKIA,{
	//...
	end;
}
```
currently (which will always be valid syntax) :
```
lighthalzen,306,267,5	script	Skia#ep162_04	4_EP16_SKIA,{
	//...
	end;

OnInit:
	cloakonnpc "Skia#ep162_04";
	end;
}
```

**Comment by Akkarinage** [comment]
> ```
> lighthalzen,306,267,5	script(CLOAKED)	Skia#ep162_04	4_EP16_SKIA,{
> 	//...
> 	end;
> }
> ```
> 
> currently (which will always be valid syntax) :
> 
> ```
> lighthalzen,306,267,5	script	Skia#ep162_04	4_EP16_SKIA,{
> 	//...
> 	end;
> 
> OnInit:
> 	cloakonnpc "Skia#ep162_04";
> 	end;
> }
> ```

Can this somehow go into script_commands.txt?

**Pull request merged**